src/v/cloud_storage_clients/s3_error.cc

```diff
@@ -420,7 +420,9 @@ rest_error_response::rest_error_response(
   std::string_view message,
   std::string_view request_id,
   std::string_view resource)
-  : _code(boost::lexical_cast<s3_error_code>(code))
+  : _code(
+      code.empty() ? s3_error_code::_unknown
+                   : boost::lexical_cast<s3_error_code>(code))
```

Comment:
should this behavior be built directly into `>>`, or is it better to check at call sites and be more defensive in our behavior?